### PR TITLE
Properly unwrap_storage tensors sent to DynamicScalar

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117444

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -4367,7 +4367,7 @@ def should_allocate(self):
 
     # TODO: handle bools carefully
     def __init__(self, sym, data):
-        super().__init__(None, NoneLayout(torch.device("cpu")), [data])  # type: ignore[arg-type]
+        super().__init__(None, NoneLayout(torch.device("cpu")), self.unwrap_storage([data]))  # type: ignore[arg-type]
```
Comment (Skylion007) [inline torch/_inductor/ir.py:4370]: Is the ignore[arg-type] still needed after the unwrap?

**Review by Skylion007 — APPROVED** [review]

File: torch/_inductor/ir.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/ir.py:4370]: Yeah, I believe the arg-type is for NoneLayout which does not inherit Layout

**Pull request closed without merging**